src/test/util/random.cpp

```diff
@@ -36,6 +38,7 @@ void SeedRandomStateForTest(SeedRand seedtype)
         return GetRandHash();
     }();
 
+    seeded_g_prng_zero = seedtype == SeedRand::ZEROS;
```

Comment:
Bug in that it can be reset if called with `FIXED_SEED` afterwards:
```suggestion
    seeded_g_prng_zero |= seedtype == SeedRand::ZEROS;
```

Wonder if it would be useful to also add checks against `FIXED_SEED` being used?

Reply:
Current behavior with a test that calls both:
```C++
SeedRandomStateForTest(SeedRand::ZEROS);
SeedRandomStateForTest(SeedRand::FIXED_SEED);
```
Resulting in `seeded_g_prng_zero` not being set... which I guess is a way of detecting that something is wrong.

My suggestion would make `seeded_g_prng_zero` _retain_ the fact that `ZEROS` had been used.

None of the above solutions detect this invalid case:
```C++
SeedRandomStateForTest(SeedRand::FIXED_SEED);
...fuzz target code using randomness...
SeedRandomStateForTest(SeedRand::ZEROS);
```

It would be more robust with my suggested change, + adding a separate `seeded_g_prng_fixed` bool which is (`|=`-)set when `FIXED_SEED` is used, so that we could error on that.